grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1049,6 +1049,51 @@ class StockMovementService {
         }
     }
 
+    void updateAdjustedItems(StockMovement stockMovement) {
+        if (stockMovement?.pickPage?.pickPageItems) {
+            stockMovement.pickPage.pickPageItems.each { PickPageItem pickPageItem ->
+                updateAdjustment(pickPageItem)
+            }
+        }
+    }
+
+    void updateAdjustedItems(PickPageItem pickPageItem) {
+        RequisitionItem requisitionItem = pickPageItem.requisitionItem
```

Comment:
in this method then you will need only to remove the picklist items and shipment items for the stock movement item and create missing picklist and shipment items (to autopick after the quantity changes)